### PR TITLE
[URGENT] Hotfix: re-adds the password and username set check in signup

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

Check for existing password and username to ensure it isn't an invite. 
If we send a team invite, the email is added to the DB and the existing check would render their sign-up process to fail as their email is in the DB.
~Reverting this to a previous state (prior to commit-1ee3783db30fe62dcf7fe1eef746759275525ecc) to check for a set password and a set username which ensures that the user is already registered and it isn't just an invite.~
Added a check to allow the previous state (if a password and username is also set), along with a check based on the fact that email is verified, which is true for SAML/SSO.

Fixes # (issue)

**Environment**: Staging(main branch)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Review by hariombalhara — APPROVED** [review]

**Review by keithwillcode — APPROVED** [review]

**Review by keithwillcode — COMMENTED** [review]
Did you mean to push up `CalendarDebug.tsx`?

**Comment by alishaz-polymath** [comment]
> Did you mean to push up `CalendarDebug.tsx`?

Definitely didn't. Wrong use of tab to auto add on my end. 😢 
Thanks for the catch!

File: apps/web/pages/api/auth/signup.ts [hunk 1]
```diff
@@ -36,14 +36,25 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
...
     where: {
       OR: [
         { username },
         {
-          AND: [{ email: userEmail }],
+          AND: [
+            { email: userEmail },
+            {
+              OR: [
+                { emailVerified: { not: null } },
```
Comment (keithwillcode) [inline apps/web/pages/api/auth/signup.ts:51]: Is this better to be a boolean check for true?

Comment (keithwillcode) [reply apps/web/pages/api/auth/signup.ts:51]: Cool.

**Review by keithwillcode — APPROVED** [review]

**Comment by keithwillcode** [comment]
Merging with failing tests because we are having issues in multiple branches/PR with E2E tests failing for reasons unassociated with code changes

**Pull request merged**